zerver/lib/markdown/__init__.py

```diff
@@ -2133,6 +2139,43 @@ def handleMatch(  # type: ignore[override] # https://github.com/python/mypy/issu
         return None, None, None
 
 
+class ImageInlineProcessor(markdown.inlinepatterns.ImageInlineProcessor):
+    def __init__(self, pattern: str, zmd: "ZulipMarkdown") -> None:
+        super().__init__(pattern, zmd)
+        self.zmd = zmd
+        # When the image is not previewable, we use this to convert it to a regular link.
+        self.link_processor = LinkInlineProcessor(pattern, zmd)
+
+    @override
+    def handleMatch(  # type: ignore[override] # https://github.com/python/mypy/issues/10197
+        self, m: Match[str], data: str
+    ) -> Tuple[Union[Element, str, None], Optional[int], Optional[int]]:
+        ret = super().handleMatch(m, data)
+        if ret[0] is not None:
+            img, match_start, index = ret
+            assert isinstance(img, Element)
+            src = img.get("src")
+            assert src is not None
+            src = sanitize_url(self.unescape(src.strip())) or ""
+            if not any(src.endswith(ext) for ext in IMAGE_EXTENSIONS):
+                # If the url is not a previewable image, gracefully fall back to a regular link
+                return self.link_processor.handleMatch(m, data)
+            # Else we wrap the image in an anchor tag and a div to enable lightbox preview.
+            alt = img.get("alt") or ""
+            a = Element("a")
+            a.set("href", src)
+            a.set("aria-label", alt)
+            src = src.lstrip("/")
```

Comment:
What?

Reply:
This is needed for thumbnailing, had not added the condition and comments explaining this yet